src/kontrol/solc_to_k.py

```diff
@@ -138,16 +163,74 @@ def flattened(self) -> list[Input]:
             return [self]
 
 
-def inputs_from_abi(abi_inputs: Iterable[dict]) -> list[Input]:
+def inputs_from_abi(abi_inputs: Iterable[dict], natspec_lengths: dict | None) -> list[Input]:
     inputs = []
-    i = 0
+    index = 0
     for input in abi_inputs:
-        cur_input = Input.from_dict(input, i)
+        cur_input = Input.from_dict(input, index, natspec_lengths)
         inputs.append(cur_input)
-        i += len(cur_input.flattened())
+        index += len(cur_input.flattened())
     return inputs
 
 
+def get_input_length(input_dict: dict, lengths: dict) -> tuple[list[int] | None, int | None]:
+    """
+    Reads from NatSpec comments the maximum length bound of an array, dynamic type, array of dynamic type, or nested arrays.
+    In case of arrays and nested arrays, the bound values are stored in a list.
+    In case of dynamic types such as `string` and `bytes` the length bound is stored in its own variable.
+    As a convention, the length of a nested array or of a dynamic type array is accessed by appending `[]` to the name of the variable.
+    i.e. for `bytes[][] _b`, the lengths are registered as:
+        _b: length of the upper most array
+        _b[]: length of the inner array
+        _b[][]: length of the `bytes` elements in the inner array.
+    If an array length is missing, the default value will be `2` to avoid generating symbolic variables.
```

Comment:
Will symbolic variables not be generated to represent the content if the length is 2?

Reply:
In that case only two variables are generated to prevent long arrays of variables.